packages/block-library/src/heading/autogenerate-anchors.js

```diff
@@ -6,7 +6,7 @@ import { isEmpty, isNil, deburr, trim } from 'lodash';
 /**
  * WordPress dependencies
  */
-import { dispatch, select, subscribe } from '@wordpress/data';
+import { select } from '@wordpress/data';
```

Comment:
Now that we're using a hook, we could name it something more clear to understand like `useGeneratedAnchor` or something like that and avoid using the global "select" and use "useSelect" instead.

Reply:
I'll need another day to figure out how `useSelect` works... 🤣 
It took me a day to understand how to use react hooks and push this refactor. Apologies, I'm a bit of a react noob so it takes me a bit longer 👍

Reply:
It's totally fine, I'm happy to help if needed :), just ping me or whatever.

Reply:
I think I got it... It works, but the code could definitely use a review 👍